doc/api/n-api.md

```diff
@@ -3395,6 +3396,147 @@ support it:
 
 <!-- it's very convenient to have all the anchors indexed -->
 <!--lint disable no-unused-definitions remark-lint-->
+## Promises
+
+N-API provides facilities for creating Promise objects as described in
+[Section 25.4][] of the ECMA specification. It implements promises as a pair of
+objects. When a promise is creaded by `napi_create_promise()`, a "deferred"
+object and the promise object itself are created. The deferred object is a
+JavaScript object tied to the promise object and must be used to resolve the
+promise by calling `napi_resolve_deferred()` or `napi_reject_deferred()`. The
+deferred is not otherwise useful, and should not be passed into JavaScript. The
+promise object can be passed into JavaScript where it can be used in the usual
+fashion.
```

Comment:
Still struggling with this a bit... perhaps something like:

```
The deferred object is a JavaScript object that is bound to the created `Promise`
and is used only to resolve or reject the `Promise` using `napi_resolve_deferred()`
or `napi_reject_deferred()`. The deferred object should not be passed into JavaScript.
The `Promise` object may be passed into JavaScript where it can be used in the usual
fashion.
```

I'm just getting hung up on that "tied" word.